### PR TITLE
Refactor/fix state sync logs

### LINKED ISSUE #514: [Refactor] Fix state sync logs
## Objective

Fix non-descriptive and inconsistent logs of consensus module's the state sync logs.

## Origin Document

Screen Shot 2023-02-15 at 15 26 09

## Goals

- Add height attributes
- Make request & response clearer (for metadata & block)

## Deliverable

- [ ] Update state sync module logs.

## Non-goals / Non-deliverables

- Change state sync business logic

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **Task specific tests or benchmarks**: `make ...`
- [ ] **New tests or benchmarks**: `make ...`
- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @gokutheengineer 
**Co-Owners**: @Olshansk

### PR DESCRIPTION
**Pull request by gokutheengineer**
## Description

Improves logs for the state sync submodule.

## Issue

Fixes #514 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [X] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Add constant state sync message types to the consensus module types
- Update logs and error messages of the state sync
- Delete unnecessary server mode enabling op in app/pocket/main.go

## Testing

- [X] `make develop_test`
- [X] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have tested my changes using the available tooling
- [X] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: consensus/debugging.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func (m *consensusModule) sendGetBlockStateSyncMessage(_ *messaging.DebugMessage
 
 	for _, val := range validators {
 		valAddress := cryptoPocket.AddressFromString(val.GetAddress())
-		if err := m.stateSync.SendStateSyncMessage(stateSyncGetBlockMessage, valAddress, requestHeight); err != nil {
-			m.logger.Debug().Msgf(typesCons.SendingStateSyncMessage(valAddress, requestHeight), err)
+		if err := m.stateSync.SendStateSyncMessage(stateSyncGetBlockMessage, typesCons.StateSyncGetBlockRequest, valAddress, requestHeight); err != nil {
+			m.logger.Error().Err(err).Msg("failed to send get block request state sync message")
```
Comment (Olshansk) [inline consensus/debugging.go:127]: 1. Option 1: s/get block request/`GetBlockRequest`
2. Option 2:
```go
m.logger.Error().Err(err)
    .Str("proto_type," "GetBlockRequest")
    .Msg("failed to send StateSyncMessage")
```

I kind of like option 2 but I'm not 100% sure if we should name this `message_type` or `proto_type` but might. Maybe worth asking in a shared channel?

Ditto elsewhere

File: consensus/state_sync/module.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ type StateSyncModule interface {
 	IsServerModEnabled() bool
 	EnableServerMode()
 
-	SendStateSyncMessage(*typesCons.StateSyncMessage, cryptoPocket.Address, uint64) error
```
Comment (Olshansk) [inline consensus/state_sync/module.go:39]: It's really weird that we're passing a string here.

1. Can you name the parameters so it's clearer to the reader
2. Let's see if we can get rid of the string

File: consensus/state_sync/module.go [hunk 3]
```diff
@@ -70,6 +70,7 @@ func (*stateSync) Create(bus modules.Bus) (modules.Module, error) {
 	}
 
 	// when node is starting, it is in sync mode, as it might need to bootstrap to the latest state
+	// TODO: consider setting this thorugh event sent by state machine for consistency
```
Comment (Olshansk) [inline consensus/state_sync/module.go:73]:
```suggestion
	// TODO: change this to to reflect the state in the fsm once merged
```

File: consensus/state_sync/module.go [hunk 4]
```diff
@@ -117,13 +118,24 @@ func (m *stateSync) EnableServerMode() {
 // TODO(#352): Implement this function
 // Placeholder function
 func (m *stateSync) HandleGetBlockResponse(blockRes *typesCons.GetBlockResponse) error {
-	m.logger.Debug().Msgf("Received get block response: %s", blockRes.Block.String())
+	consensusMod := m.GetBus().GetConsensusModule()
+	serverNodePeerId := consensusMod.GetNodeAddress()
+	clientPeerId := blockRes.PeerAddress
+
+	m.logger.Info().Msgf("%s received get block response from: %s, for height %d. Received block's header is: %s,  \n", serverNodePeerId, clientPeerId, blockRes.Block.BlockHeader.Height, blockRes.Block.BlockHeader)
```
Comment (Olshansk) [inline consensus/state_sync/module.go:125]: 1. try to use `GetGlockResponse` (the full proto, type or var name) where appropriate. It'll make it much easier to read in the logs.
2. You have lots of  parameters here (height, heater, etc...). Look at how you can set fields: https://github.com/pokt-network/pocket/wiki/Logger-Readme#fields

File: consensus/types/errors.go [hunk 5]
```diff
@@ -83,8 +83,8 @@ func SendingMessage(msg *HotstuffMessage, nodeId NodeId) string {
 	return fmt.Sprintf("✉️ Sending message ✉️ to %d at (height, step, round) (%d, %d, %d)", nodeId, msg.Height, msg.Step, msg.Round)
 }
 
-func SendingStateSyncMessage(nodeId cryptoPocket.Address, height uint64) string {
```
Comment (Olshansk) [inline consensus/types/errors.go:86]: I think we need to rename this file from `errors.go` to `messages.go`

File: consensus/state_sync/helpers.go [hunk 6]
```diff
@@ -6,12 +6,12 @@ import (
...
 )
 
-func (m *stateSync) SendStateSyncMessage(stateSyncMsg *typesCons.StateSyncMessage, peerId cryptoPocket.Address, blockHeight uint64) error {
+func (m *stateSync) SendStateSyncMessage(stateSyncMsg *typesCons.StateSyncMessage, msgType string, peerId cryptoPocket.Address, blockHeight uint64) error {
 	anyMsg, err := anypb.New(stateSyncMsg)
 	if err != nil {
 		return err
 	}
-	m.logger.Info().Uint64("height", blockHeight).Msg(typesCons.SendingStateSyncMessage(peerId, blockHeight))
+	m.logger.Info().Uint64("height", blockHeight).Msg(typesCons.SendingStateSyncMessage(peerId, msgType, blockHeight))
```
Comment (Olshansk) [inline consensus/state_sync/helpers.go:14]: In `types/message.go`, we use `ProtoReflect().Descriptor().Name()`

```go
func getMessageType(msg Message) string {
	return string(msg.ProtoReflect().Descriptor().Name())
}
```

I think you might be able to use this and remove the msgType string altogether.

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: consensus/types/errors.go (hunk 5 above)
Comment (gokutheengineer) [reply consensus/types/errors.go:86]: makes sense, thanks.

File: consensus/state_sync/module.go (hunk 4 above)
Comment (gokutheengineer) [reply consensus/state_sync/module.go:125]: thanks for the suggestions!

File: consensus/debugging.go (hunk 1 above)
Comment (gokutheengineer) [reply consensus/debugging.go]: I agree that proto_type makes sense here, as message_type can be confusing with the HotStuffMessage, StateSyncMessage etc.proto

File: consensus/state_sync/helpers.go (hunk 6 above)
Comment (gokutheengineer) [reply consensus/state_sync/helpers.go]: Using this way returns StateSync, and not the type of the oneOf field, i.e. GetBlockRequest etc. Therefore I implemented getMessageType function as:

```go
	switch msg.Message.(type) {
	case *typesCons.StateSyncMessage_MetadataReq:
		return "MetadataRequest"
	case ...
	}
```

File: consensus/state_sync/server.go [hunk 7]
```diff
@@ -60,14 +67,20 @@ func (m *stateSync) HandleStateSyncMetadataRequest(metadataReq *typesCons.StateS
...
+	lastPersistedBlockHeight := consensusMod.CurrentHeight() - 1
+
+	fields := map[string]interface{}{
+		"height":   lastPersistedBlockHeight,
+		"sender":   serverNodePeerAddress,
+		"receiver": clientPeerAddress,
+	}
 
-	currentHeight := m.GetBus().GetConsensusModule().CurrentHeight()
+	m.logger.Info().Fields(fields).Msgf("Received state sync get block request: %s", blockReq)
```
Comment (Olshansk) [inline consensus/state_sync/server.go:80]:
```suggestion
	m.logger.Info().Fields(fields).Msgf("Received StateSync GetBlockRequest: %s", blockReq)
```

File: consensus/state_sync/server.go [hunk 8]
```diff
@@ -22,13 +22,20 @@ type StateSyncServerModule interface {
...
+	// current height is the height of the block that is being processed, so we need to subtract 1 for the last finalized block
+	lastPersistedBlockHeight := consensusMod.CurrentHeight() - 1
+
+	fields := map[string]interface{}{
+		"height":   lastPersistedBlockHeight,
+		"sender":   serverNodePeerAddress,
+		"receiver": clientPeerAddress,
+	}
+
+	m.logger.Info().Fields(fields).Msgf("Received state sync get block request: %s", metadataReq)
```
Comment (Olshansk) [inline consensus/state_sync/server.go:36]:
```suggestion
	m.logger.Info().Fields(fields).Msgf("Received StateSync Metadata %s", metadataReq)
```

File: consensus/state_sync/module.go [hunk 9]
```diff
@@ -117,13 +118,36 @@ func (m *stateSync) EnableServerMode() {
...
+	clientPeerId := metaDataRes.PeerAddress
+	currentHeight := consensusMod.CurrentHeight()
+
+	fields := map[string]interface{}{
+		"currentHeight": currentHeight,
+		"sender":        serverNodePeerId,
+		"receiver":      clientPeerId,
+	}
+
+	m.logger.Info().Fields(fields).Msgf("Received get metadata response: %s", metaDataRes)
```
Comment (Olshansk) [inline consensus/state_sync/module.go:150]:
```suggestion
	m.logger.Info().Fields(fields).Msgf("Received StateSync MetadataResponse: %s", metaDataRes)
```

File: consensus/state_sync/module.go [hunk 10]
```diff
@@ -117,13 +118,36 @@ func (m *stateSync) EnableServerMode() {
...
+	serverNodePeerId := consensusMod.GetNodeAddress()
+	clientPeerId := blockRes.PeerAddress
+
+	fields := map[string]interface{}{
+		"currentHeight": blockRes.Block.BlockHeader.Height,
+		"sender":        serverNodePeerId,
+		"receiver":      clientPeerId,
+	}
+
+	m.logger.Info().Fields(fields).Msgf("Received get block response: %s", blockRes)
```
Comment (Olshansk) [inline consensus/state_sync/module.go:131]:
```suggestion
	m.logger.Info().Fields(fields).Msgf("Received StateSync GetBlockResponse: %s", blockRes)
```

**Review by Olshansk — CHANGES_REQUESTED** [review]
@gokutheengineer Small suggestion on code style and also trying to stay consistent with the log messages.

The way you're using the fields is 💯 . Curious if you got benefit out of printing the entire proto request/response?

File: consensus/debugging.go [hunk 11]
```diff
@@ -149,9 +151,11 @@ func (m *consensusModule) sendGetMetadataStateSyncMessage(_ *messaging.DebugMess
 	}
 
 	for _, val := range validators {
-		valAddress := cryptoPocket.AddressFromString(val.GetAddress())
-		if err := m.stateSync.SendStateSyncMessage(stateSyncMetaDataReqMessage, valAddress, requestHeight); err != nil {
-			m.logger.Debug().Msgf(typesCons.SendingStateSyncMessage(valAddress, requestHeight), err)
+		if m.GetNodeAddress() != val.GetAddress() {
```
[the remark this answers is not in the document]
Comment (gokutheengineer) [reply consensus/debugging.go:155]: Done, thanks!

**Comment by gokutheengineer** [comment]
> Curious if you got benefit out of printing the entire proto request/response?

Definitely, it makes it easier to read logs.

File: consensus/state_sync/module.go [hunk 12]
```diff
@@ -118,13 +119,36 @@ func (m *stateSync) EnableServerMode() {
 // TODO(#352): Implement this function
 // Placeholder function
 func (m *stateSync) HandleGetBlockResponse(blockRes *typesCons.GetBlockResponse) error {
-	m.logger.Debug().Msgf("Received get block response: %s", blockRes.Block.String())
+	consensusMod := m.GetBus().GetConsensusModule()
+	serverNodePeerId := consensusMod.GetNodeAddress()
+	clientPeerId := blockRes.PeerAddress
+
+	fields := map[string]interface{}{
```
Comment (Olshansk) [inline consensus/state_sync/module.go:126]: NIT: Replace `interface{}` with `any`. Ditto below

**Review by Olshansk — APPROVED** [review]
Just one last minor comment but otherwise, LGTM!

**Pull request merged**